lib/monitor/proxy_test.go

```diff
@@ -80,69 +80,67 @@ func TestProxy(t *testing.T) {
 	}
 }
 
-// XXX Running both tests sequentially on travis fails.
-// Issue reported. Meanwhile, we jsut comment that out ,because the first test
-// is already testing the same functionality.
-/*func TestReadyProxy(t *testing.T) {*/
-//dbg.TestOutput(testing.Verbose(), 3)
-//m := make(map[string]string)
-//m["machines"] = "1"
-//m["ppm"] = "1"
-//stat := NewStats(m)
-//// First set up monitor listening
-//monitor := NewMonitor(stat)
-//defer monitor.Stop()
-//done := make(chan bool)
-//go func() {
-//err := monitor.Listen()
-//if err != nil {
-//monitor.Stop()
-//t.Fatal(fmt.Printf("Error listening monitor %v", err))
-//}
-//done <- true
-//}()
-//time.Sleep(200 * time.Millisecond)
-//// Then setup proxy
-//// change port so the proxy does not listen to the same
-//// than the original monitor
-//oldSink := SinkPort
-//SinkPort = 8000
-//// proxy listen to 0.0.0.0:8000 & redirect to
-//// localhost:4000
-//go func() {
-//err := Proxy("localhost:" + strconv.Itoa(oldSink))
-//if err != nil {
-//t.Fatal(fmt.Printf("Error connecting proxy %v", err))
-//}
-//}()
+func TestReadyProxy(t *testing.T) {
+	dbg.TestOutput(testing.Verbose(), 3)
+	m := make(map[string]string)
+	m["machines"] = "1"
+	m["ppm"] = "1"
+	stat := NewStats(m)
+	// First set up monitor listening
+	monitor := NewMonitor(stat)
+	done := make(chan bool)
+	go func() {
+		if err := monitor.Listen(); err != nil {
+			dbg.Error("Could not start listener:", err)
+			dbg.Error("Retry once in 500ms ...")
+			time.Sleep(500 * time.Millisecond)
+			if err := monitor.Listen(); err != nil {
+				dbg.Fatal("Could not start listener:", err)
+			}
+		}
```

Comment:
Does it work without this? Will say, is that a remnant of a try to make it work, or is it needed? 
If needed {why? Is it not possible to make TestProxy wait for correct closing of the ports?} else { remove }